src/hyperion/external_interaction/callbacks/xray_centre/ispyb_callback.py

```diff
@@ -43,6 +42,21 @@
     from event_model import Event, RunStart, RunStop
 
 
+def ispyb_activation_wrapper(parameters, plan_generator):
+    @bpp.run_decorator(
+        md={
+            "activate_callbacks": ["GridscanISPyBCallback"],
+            "subplan_name": CONST.PLAN.GRID_DETECT_AND_DO_GRIDSCAN,
+            "hyperion_internal_parameters": parameters.json(),
+        }
+    )
+    @wraps(plan_generator)
+    def wrapped():
+        return plan_generator
+
+    yield from wrapped()
```

Comment:
Should: 
```suggestion
def ispyb_activation_wrapper(parameters, plan_generator):
    return bpp.run_wrapper(
        plan_generator,
        md={
            "activate_callbacks": ["GridscanISPyBCallback"],
            "subplan_name": CONST.PLAN.GRID_DETECT_AND_DO_GRIDSCAN,
            "hyperion_internal_parameters": parameters.json(),
        },
    )
```

Reply:
Nit: I think you could then switch the `parameters` and `plan_generator` to more closely match the `run_wrapper` syntax